### PR TITLE
Add Gleam parser and queries

### PR DESCRIPTION
**Pull request by connorlay**
Relates to: https://github.com/J3RN/tree-sitter-gleam/issues/1 and https://github.com/J3RN/tree-sitter-gleam/issues/12

cc @lpil @J3RN

This PR adds support for the [Gleam](https://gleam.run/) language via [tree-sitter-gleam](https://github.com/J3RN/tree-sitter-gleam). I have not measured it, but the highlighting performance is excellent on my machine. The queries are based on those in the parser's repository, adapted to use nvim-treesitter's additional capture groups.

image
An example web server taken from the Gleam homepage. Color theme is [Melange](https://github.com/savq/melange).

### CODE REVIEW

File: queries/gleam/highlights.scm
Comment (theHamsta) [inline queries/gleam/highlights.scm:117]: maybe event `@constant.builtin`

File: queries/gleam/highlights.scm
Comment (connorlay) [inline queries/gleam/highlights.scm:166]: @theHamsta Is it possible to define highlights that take into account the scope of a node? Consider the following:
```gleam
pub fn foo() {
  fizz.buzz()
}

pub fn foo(bar) {
  bar.baz()
}
```
`fizz` should be highlighted as `@namespace` since it is not defined anywhere in the function parameters or body. However, `bar` should be highlighted as `@variable` because it is defined within the function parameter list.

Thanks 😄

Comment (theHamsta) [reply queries/gleam/highlights.scm:166]: yes, this is possible using `(#is? @variable parameter)`. But our `#is?` implementation is super slow. It's waiting for something like https://github.com/nvim-treesitter/nvim-treesitter/pull/1120 but something that is preferable something faster that `:parent()`

Comment (connorlay) [reply queries/gleam/highlights.scm:166]: Good to know, I'll add a note about this in the highlights.scm and we can revisit it later. Thanks!

File: queries/gleam/indents.scm
Comment (theHamsta) [inline queries/gleam/indents.scm:21]: Again, tests would be helpful

File: queries/gleam/highlights.scm
Comment (theHamsta) [inline queries/gleam/highlights.scm:10]: Is try also more in direction of `@exception` or more like "launch" or general control flow?

**Review by theHamsta — APPROVED** [review]

**Comment by connorlay** [comment]
Let's hold off on merging this until I have some tests in place. Thanks!

**Review by theHamsta — APPROVED** [review]

File: queries/gleam/highlights.scm
Comment (lpil) [reply queries/gleam/highlights.scm:10]: It's flow control, similar to this Go code:

```go
if err != nil {
        return err
}
```

File: lua/nvim-treesitter/parsers.lua
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:335]: You would be the maintainer for nvim-treesitter for this parser.

Comment (connorlay) [reply lua/nvim-treesitter/parsers.lua:335]: For the maintainers section, is it intended to be the maintainer of the queries and tests in nvim-treesitter, or the maintainer of the parser itself? For example, I am the maintainer for tree-sitter-heex and the repository lives under my namespace, but I did not write tree-sitter-gleam.

File: lua/nvim-treesitter/parsers.lua
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:335]: @connorlay does J3RN also contribute to nvim-treesitter? Otherwise, he should be removed from the maintainers as I and user otherwise will unnecessarily ping him when there are nvim-treesitter issues

Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:335]: otherwise this PR is fine

Comment (connorlay) [reply lua/nvim-treesitter/parsers.lua:335]: No J3RN is the maintainer of the tree-sitter-gleam parser itself, but not the nvim-treesitter integration.

**Comment by connorlay** [comment]
@theHamsta I updated the maintainers list as you requested. I think this is ready to merge 😄

**Comment by theHamsta** [comment]
Thank you a lot for your help!

**Pull request merged**
